### PR TITLE
Search Improvements

### PR DESCRIPTION
**Pull request by LucasCairns**

### CODE REVIEW

File: moj-node/test/repositories/SearchRepository.spec.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        });
+
+        const repository = searchRepository(client);
+
+        const response = await repository.find({
+          query: 'Test Query',
+          prison: 'HMP Development',
+        });
+
+        expect(client.post.callCount).to.equal(
```
Comment (romidane) [inline moj-node/test/repositories/SearchRepository.spec.js]: You probably want to test that that the client was called with the query provided?

File: moj-node/test/repositories/SearchRepository.spec.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        });
+
+        const repository = searchRepository(client);
+
+        await repository.find({
+          query: 'Test Query',
+          prison: 'HMP Development',
+        });
+
+        expect(client.post.callCount).to.equal(
```
Comment (romidane) [inline moj-node/test/repositories/SearchRepository.spec.js]: don't need to test this each time?

File: moj-node/test/repositories/SearchRepository.spec.js [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        );
+
+        const requestBody = client.post.lastCall.args[1];
+
+        expect(requestBody.query.bool.should[0].term.prison_name).to.equal(
+          'HMP Development',
+          'The query should filter by prison name',
+        );
+
+        expect(
```
Comment (romidane) [inline moj-node/test/repositories/SearchRepository.spec.js]: Don't do this... your coupling the test with the with the search provider.. just check that that the `stringified` request body contains the prison name

File: moj-node/test/repositories/SearchRepository.spec.js (hunk 1 above)
Comment (LucasCairns) [reply moj-node/test/repositories/SearchRepository.spec.js]: Added assertions to check for the query in the request body

File: moj-node/test/repositories/SearchRepository.spec.js (hunk 3 above)
Comment (LucasCairns) [reply moj-node/test/repositories/SearchRepository.spec.js]: Stringified the requests body and added assertions for the prison name in the body

File: moj-node/test/repositories/SearchRepository.spec.js (hunk 2 above)
Comment (LucasCairns) [reply moj-node/test/repositories/SearchRepository.spec.js]: Going to leave as is, we're making assertions on the args, I feel the message would be more meaningful if the test were to break

File: moj-node/cypress/e2e/homepage.spec.js [hunk 4]
```diff
@@ -9,15 +9,15 @@ describe('Homepage', () => {
...
           .should('contain', 'Home')
-          .should('contain', 'Working in Berwyn')
+          .should('contain', 'Education and work')
           .should('contain', 'Getting out');
       });
 
       it(`navigates to the 'Working in Berwyn' page`, () => {
         const workingInBerwynLink = `${cssNav} > :nth-child(2) > .govuk-link`;
 
         cy.get(workingInBerwynLink).click();
```
Comment (romidane) [inline moj-node/cypress/e2e/homepage.spec.js]: Probably should rename this variable?

File: moj-node/server/repositories/search.js [hunk 5]
```diff
@@ -29,22 +39,32 @@ function searchRepository(httpClient) {
...
-          .prefixLength(2)
-          .operator('and'),
+          .should([
+            esb.termQuery('prison_name', prison),
+            esb.boolQuery().mustNot([esb.existsQuery('prison_name')]),
+          ])
+          .minimumShouldMatch(1),
       )
       .size(limit)
       .timeout('15ms')
```
Comment (romidane) [inline moj-node/server/repositories/search.js]: Have we tried increasing the timeout to see if yields better results?
for example we get nothing after the 4th character.. maybe increasing the timeout might yield results?

File: moj-node/server/routes/search.js [hunk 6]
```diff
@@ -10,17 +9,17 @@ module.exports = function createSearchRouter({ searchService, logger }) {
...
     logger.info('GET /search');
 
+    const { establishmentId } = req.app.locals.envVars;
+
     try {
       let results = [];
-      const query = req.query.search;
+      const { query } = req.query;
 
       if (query) {
```
Comment (romidane) [inline moj-node/server/routes/search.js]: This logic could live in the service later
`  if (query) {`
Cleaning up the route from logic

File: moj-node/server/services/search.js [hunk 7]
```diff
@@ -1,10 +1,17 @@
...
+  getEstablishmentName = utils.getEstablishmentName,
+}) {
+  function find({ query, limit, from, establishmentId }) {
+    const prison = getEstablishmentName(establishmentId);
+    return searchRepository.find({ query, limit, from, prison });
   }
 
-  function typeAhead({ query, limit }) {
-    return searchRepository.typeAhead({ query, limit });
+  function typeAhead({ query, limit, establishmentId }) {
```
Comment (romidane) [inline moj-node/server/services/search.js]: let's add a check to see if we have a query here?

File: moj-node/server/services/search.js [hunk 8]
```diff
@@ -1,10 +1,17 @@
-module.exports = function createSearchService({ searchRepository }) {
-  function find({ query, limit, from }) {
-    return searchRepository.find({ query, limit, from });
+const utils = require('../utils');
+
+module.exports = function createSearchService({
+  searchRepository,
+  getEstablishmentName = utils.getEstablishmentName,
+}) {
+  function find({ query, limit, from, establishmentId }) {
```
Comment (romidane) [inline moj-node/server/services/search.js]: let's add a check to see if we have a query here?

File: moj-node/server/views/components/search/template.njk [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<div  id="search-wrapper">
+  <form method="GET" action="/search">
+    <label class="govuk-label" for="event-name">Search</label>
+    <input class="govuk-search-input search-box" id="search" name="query" type="text" value="{{params.query}}" autocomplete="off" placeholder="Search the content hub">
```
Comment (romidane) [inline moj-node/server/views/components/search/template.njk]: Is the placeholder text Nigel approved 😅 ?

File: moj-node/server/views/pages/search.html [hunk 10]
```diff
@@ -26,6 +26,34 @@
   <![endif]-->
 {% endblock %}
 
+{% block analytics %}
+<!-- Matomo -->
+<script>
```
Comment (romidane) [inline moj-node/server/views/pages/search.html]: this block should be deleted in this file
it's now `{%block matomo %} which is further down the file`

File: moj-node/cypress/e2e/homepage.spec.js [hunk 11]
```diff
@@ -9,15 +9,15 @@ describe('Homepage', () => {
       it('contains the correct items in the navigation', () => {
         cy.get(cssNav)
           .should('contain', 'Home')
-          .should('contain', 'Working in Berwyn')
+          .should('contain', 'Education and work')
           .should('contain', 'Getting out');
       });
 
       it(`navigates to the 'Working in Berwyn' page`, () => {
```
Comment (romidane) [inline moj-node/cypress/e2e/homepage.spec.js]: Probably also worth updating the test title here?

File: moj-node/server/views/components/search/template.njk (hunk 9 above)
Comment (LucasCairns) [reply moj-node/server/views/components/search/template.njk]: Yes

File: moj-node/cypress/e2e/homepage.spec.js (hunk 4 above)
Comment (LucasCairns) [reply moj-node/cypress/e2e/homepage.spec.js]: Done

File: moj-node/server/routes/search.js (hunk 6 above)
Comment (LucasCairns) [reply moj-node/server/routes/search.js]: Moved into service

File: moj-node/server/services/search.js (hunk 8 above)
Comment (LucasCairns) [reply moj-node/server/services/search.js]: Resolved, added assertion for query string

File: moj-node/server/services/search.js (hunk 7 above)
Comment (LucasCairns) [reply moj-node/server/services/search.js]: Resolved, added assertion for query string

File: moj-node/cypress/e2e/homepage.spec.js (hunk 11 above)
Comment (LucasCairns) [reply moj-node/cypress/e2e/homepage.spec.js]: Done

File: moj-node/server/views/pages/search.html (hunk 10 above)
Comment (LucasCairns) [reply moj-node/server/views/pages/search.html]: Done

File: moj-node/server/repositories/search.js (hunk 5 above)
Comment (LucasCairns) [reply moj-node/server/repositories/search.js]: Removed the timeout, we have such a small dataset that we're likely to resolve in a reasonable time querying it's entirety. The results remain the same, it's likely we need to experiment with prefix-lengths and fuzziness.

**Pull request merged**